This is the Travis failure:

<details>
<summary>

<code>[09/Sep/2016 18:06:19] "GET /static/img/logos/psf.png HTTP/1.1" 404 98</code></summary>



```
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/django_amber/management/commands/buildsite.py", line 20, in handle
    self.buildsite(port)
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/django_amber/management/commands/buildsite.py", line 31, in buildsite
    rsp.raise_for_status()
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/requests/models.py", line 844, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://localhost:54234/static/img/logos/psf.png
make[1]: *** [build] Error 1
```

</details>

I can’t seem to reproduce this error locally – when I check out this branch and run `./pre-flight-checks.sh`, it exits cleanly. Is it Travis with the dodgy setup, or me?

--- a/pre-flight-checks.sh
+++ b/pre-flight-checks.sh
@@ -8,25 +8,12 @@ ERRORS=()
 
 echo " *** Checking that site can be built."
 
-python manage.py serve >/dev/null 2>&1 &
-SERVER_PID=$!
+make build
 
 if [[ $? -eq 0 ]]; then
 	echo " *** Site built ok."
 else
 	echo " *** Site could not be built."
-	exit 1
-fi
-
-###
-# Wait for server to start.
-###
-
-sleep 5
-curl "http://localhost:8000" >/dev/null 2>&1
-
-if [[ $? -ne 0 ]]; then
-	echo " *** Server is not running."
 	exit 1
 fi
 
@@ -55,8 +42,6 @@ grep -e "Pycon UK" -e "pycon UK" -e "pyconUK" -e "PyConUK" --line-number --recur
 if [[ $? -eq 0 ]]; then
 	ERRORS+=("Conference name is not spelt correctly")
 fi
-
-kill $SERVER_PID
 
 if [[ ${#ERRORS[@]} -eq 0 ]]; then
 	echo " *** All pre-flight checks passed!"

--- a/pyconuk/settings.py
+++ b/pyconuk/settings.py
@@ -19,11 +19,18 @@ BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/1.9/howto/deployment/checklist/
 
-# SECURITY WARNING: keep the secret key used in production secret!
+# This does not need to be kept secret, since it is not used to protect
+# anything in the static site.
 SECRET_KEY = 'secret'
 
-# SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = True
+# SECURITY WARNING: This should be False when the site is built to ensure we
+# don't accidentally leak information in error pages.  This has the added
+# effect of massively speeding up the build, since LESS compilation no longer
+# happens on every request!
+DEBUG = bool(os.getenv('DEBUG', False))
+
+# This is fine.
+ALLOWED_HOSTS = ['*']
 
 
 # Application definition

--- a/pyconuk/utils.py
+++ b/pyconuk/utils.py
@@ -86,7 +86,6 @@ def compute_html_table_dimensions(table):
         if row:
             table_with_dimensions.append(row)
 
-    print(table_with_dimensions)
     return table_with_dimensions
 
 

